### PR TITLE
Homepage UI Improvements

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor (I guess?)
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
It's basically frontend work with low-cost and low-effort UI changes. Improving contrast, cleaning up the feed, de-emphasizing sidebars a bit, etc. 

About about "why" can be found here:
- https://dev.to/devteam/hello-again-time-to-do-some-work-3b6o
- https://dev.to/devteam/follow-up-your-feedback-on-the-home-page-2cij

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
before vs after
(left is current, right is new)

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/views/articles/_sidebar_nav.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="sidebar-nav" id="sidebar-nav">
   <% if user_signed_in? %>
-    <div class="sidebar-profile-snapshot">
+    <div class="sidebar-profile-snapshot" style="display: none;"> <%# TODO: i'd rather get rid of this whole section, but removing it actually breaks some JS :( %>
```
Comment (benhalpern) [inline app/views/articles/_sidebar_nav.html.erb]: I think in the spirit of frog boiling, we might want to keep this around and do a full removal as a future step.

File: app/assets/stylesheets/_mixins.scss [hunk 2]
```diff
@@ -44,3 +44,17 @@
     color: inherit
   }
 }
+
+
+
+// Card styling
+@mixin dev-card {
```
Comment (benhalpern) [inline app/assets/stylesheets/_mixins.scss]: I'm getting extra thick borders on the article show page in the default styling (not with the themes). Not sure if that's intentional.

File: app/assets/javascripts/utilities/buildArticleHTML.js.erb [hunk 3]
```diff
@@ -49,7 +49,7 @@ function buildArticleHTML(article) {
     var profileUsername = article.user.username
     var orgHeadline = "";
     if (article.organization && !document.getElementById("organization-article-index")) {
-      orgHeadline = '<div class="article-organization-headline"><a class="org-headline-filler" href="/'+article.organization.slug+'"><span class="article-organization-headline-inner"><img alt="'+article.organization.name+' logo" src="'+article.organization.profile_image_90+'" loading="lazy">'+article.organization.name+'</span></a></div>'
+      orgHeadline = '<div class="article-organization-headline"><a class="org-headline-filler" href="/'+article.organization.slug+'"><img alt="'+article.organization.name+' logo" src="'+article.organization.profile_image_90+'" loading="lazy">'+article.organization.name+'</a></div>'
```
Comment (benhalpern) [inline app/assets/javascripts/utilities/buildArticleHTML.js.erb]: This is implemented here and also in the server at `app/views/articles/_single_story.html.erb` and a similar card is rendered in notifications... `app/views/notifications/_article.html.erb`

File: app/assets/stylesheets/articles.scss [hunk 4]
```diff
@@ -4,301 +4,73 @@
 @import 'variables';
 @import '_mixins';
 
+
 .home {
   position: relative;
-  min-height: 440px;
-  margin: auto;
+  min-height: 440px; // TODO: why?
+  margin: 0 auto;
```
Comment (benhalpern) [inline app/assets/stylesheets/articles.scss]: There are some styles in `_top.html.erb` which can be removed. They were added due to a weird css specificity bug that only occurred in production so we quickly hacked it in.

It can be removed and if we happen to encounter the same bug again we can fix it again. 🤷‍♂

File: app/assets/stylesheets/articles.scss [hunk 5]
```diff
@@ -471,20 +208,20 @@
...
+        width: calc(100% - 72px);
+        box-sizing: border-box; // TODO: this should be applied globally, not only .content
+        padding: 16px 16px 100px 16px; // TODO: my, oh my...
         h3 {
           margin: 0px;
-          font-size: 21px;
-          line-height: 25px;
-          width: 100%;
+          font-size: 20px;
+          font-weight: 700;
```
Comment (benhalpern) [inline app/assets/stylesheets/articles.scss]: Should we also change the article title font weight to `700` in this PR? Even though this is primarily front-page-focused, we should ensure we keep other areas consistent eh?

**Comment by benhalpern** [comment]
)

This CTA is messed up

File: app/views/articles/_sidebar_nav.html.erb (hunk 1 above)
Comment (ludwiczakpawel) [reply app/views/articles/_sidebar_nav.html.erb]: Good point, I'll bring it back.

File: app/assets/stylesheets/_mixins.scss (hunk 2 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/_mixins.scss]: Definitely not. I'll take a look. I still have to QA other views.

File: app/assets/javascripts/utilities/buildArticleHTML.js.erb (hunk 3 above)
Comment (ludwiczakpawel) [reply app/assets/javascripts/utilities/buildArticleHTML.js.erb]: I've just checked - I have actually changed it in these files already. Is there any specific reason why you brought it up?

File: app/assets/stylesheets/articles.scss (hunk 4 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: Since this PR is (and will) not be about  actual CSS refactoring primarily, I'd rather keep it as it is for now and make sure it's not a _breaking change_ in some other PR. That `// TODO: ...` was more of a note to myself to check it in future.

File: app/views/articles/_sidebar_nav.html.erb (hunk 1 above)
Comment (ludwiczakpawel) [reply app/views/articles/_sidebar_nav.html.erb]: Consider it done.

File: app/assets/stylesheets/_mixins.scss (hunk 2 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/_mixins.scss]: I fixed this one and couple more things on this view. Also few other "cards" stylings here and there...

**Comment by ludwiczakpawel** [comment]
The CTAs, other issues AND some extra ones should now be solved. I'll give it another pass tomorrow as well.

**Comment by benhalpern** [comment]
This is looking pretty great.

I notice a few visual glitches and edge cases, but all small things. There are also some tests which need to be adjusted and the conflicts to be resolved.

So once you're at a good place with this I can fix the edge cases and tests etc. myslelf and push to this branch. I think that's easier than trying to explain each one.

How's that sound? Let me know when you are ready to let this bird out of the cage.

**Comment by ludwiczakpawel** [comment]
If we're not in rush, I'll take it for couple more rides by the end of the week and then you can move on with your fixes.

**Comment by benhalpern** [comment]
> If we're not in rush, I'll take it for couple more rides by the end of the week and then you can move on with your fixes.

@ludwiczakpawel sounds good to me!

**Comment by ludwiczakpawel** [comment]
Ok, I think for now I'm pretty much done with changes... It's already quite a lot for one deploy :D

File: app/assets/stylesheets/article-show.scss [hunk 6]
```diff
@@ -886,26 +868,15 @@ article {
...
-    @include themeable(
-      background,
-      theme-background,
-      $tan
-    );
+    @include themeable(background, theme-container-background, $tan);
+    @include themeable(border, theme-container-border, none);
+    @include themeable(box-shadow, theme-container-box-shadow, $bold-shadow);
+
+    @media screen and (min-width: 950px) {
```
Comment (Ridhwana) [inline app/assets/stylesheets/article-show.scss]: probably for the future and not for now but it would be cool if we could add our media query breakpoints to variables and define them upfront instead of using the numbers everywhere,

File: app/assets/stylesheets/article-show.scss [hunk 7]
```diff
@@ -886,26 +868,15 @@ article {
...
+      border-radius: 3px 3px 0 0;
```
Comment (Ridhwana) [inline app/assets/stylesheets/article-show.scss]: small comment, you can maybe clean this up with just 3px 0;

File: app/assets/stylesheets/footer.scss [hunk 8]
```diff
@@ -35,23 +35,24 @@ footer {
...
+      padding: 48px 32px;
+      box-sizing: border-box;
       line-height: 3em;
       margin: auto;
+      @include themeable(background, theme-container-background, $tan);
+      @include themeable(border, theme-container-border, none);
+      @include themeable(box-shadow, theme-container-box-shadow, $bold-shadow);
+
+      @media screen and (min-width: 950px) {
+        border-radius: 3px 3px 0 0;
```
Comment (Ridhwana) [inline app/assets/stylesheets/footer.scss]: also here: 3px 0;

File: app/views/users/_main_feed.html.erb [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <% if @pinned_stories.present? %>
   <div class="pinned-articles">
-    <div class="pinned-articles-indicator"><img src="<%= asset_path("pushpin.svg") %>" alt="pin" /> Pinned</div>
+    <h5><img src="<%= asset_path("pushpin.svg") %>" width="20" height="20" alt="pin" /> Pinned</h5>
```
Comment (Ridhwana) [inline app/views/users/_main_feed.html.erb]: should we rather add a class here instead of inline width, height css?

File: app/assets/javascripts/initializers/initScrolling.js.erb [hunk 10]
```diff
@@ -73,7 +73,7 @@ function buildTagsHTML(tag) {
     negativeFollow = 0;
   }
   return tagHTML +
-        '<div id="follows-' + tag.id + '" class="single-article" style="border-color:' + tag.color + ';box-shadow: 3px 3px 0px ' + tag.color + '">\n' +
+        '<div id="follows-' + tag.id + '" class="single-article" style="border-color: 1px solid ' + tag.color + ';box-shadow: 3px 3px 0px ' + tag.color + '">\n' +
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/javascripts/initializers/initScrolling.js.erb]: whoops, good catch!

File: app/assets/javascripts/initializers/initializeBaseUserData.js [hunk 11]
```diff
@@ -3,20 +3,14 @@
 /* global userData, filterXSS */
 
 function initializeUserProfileContent(user) {
-  document.getElementById('sidebar-profile-pic').innerHTML =
-    '<img alt="' +
-    user.username +
-    '" class="sidebar-profile-pic-img" src="' +
-    user.profile_image_90 +
-    '" />';
+  document.getElementById('sidebar-profile--avatar').src = user.profile_image_90;
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/javascripts/initializers/initializeBaseUserData.js]: 100% agree. out of scope for now though. what i did here is basically renamend some of them so they make a bit more sense. But [I definitely agree](https://dev-internal.slack.com/archives/C7GE84DEJ/p1579172524145300) we should use classes. Probably gonna be a separate PR while refactoring pieces of code.

File: app/assets/stylesheets/article-show.scss (hunk 6 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/article-show.scss]: 100% agree. But for future. Refactoring some bits and pieces of CSS is on my plate, but again - definitely separate PR(s)

File: app/assets/javascripts/initializers/initializeBaseUserData.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (Ridhwana) [reply app/assets/javascripts/initializers/initializeBaseUserData.js]: Nice!

File: app/views/users/_main_feed.html.erb (hunk 9 above)
Comment (ludwiczakpawel) [reply app/views/users/_main_feed.html.erb]: Defining width and height is good convention: if CSS is not (fully) loaded the image can otherwise see huge and cause unnecessary layout jump.

Comment (Ridhwana) [reply app/views/users/_main_feed.html.erb]: cool - sounds good, didn't think of that!

File: app/assets/stylesheets/article-show.scss (hunk 7 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/article-show.scss]: `border-radius: 3px 0;` is a shorthand for `border-radius: 3px 0 3px 0;` so it's different than what I have - I know, CSS is weird... :)

Comment (Ridhwana) [reply app/assets/stylesheets/article-show.scss]: haha i actually realised that after i posted it and deleted the comment, but clearly it did not delete!

**Comment by ludwiczakpawel** [comment]
@Ridhwana  this is very good feedback, thank you!

I've resolved one issue with border you pointed out. Otherwise, you touch on some refactoring aspects - I think it's worth doing it, but I'd scope it as separate project/PR. This one is already quite big and is not really about optimizing/refactoring code. But, I definitely agree this should be done and I put some of it on my plate already.

**Comment by Ridhwana** [comment]
> @Ridhwana this is very good feedback, thank you!
> 
> I've resolved one issue with border you pointed out. Otherwise, you touch on some refactoring aspects - I think it's worth doing it, but I'd scope it as separate project/PR. This one is already quite big and is not really about optimizing/refactoring code. But, I definitely agree this should be done and I put some of it on my plate already.

Awesomeness, yeah I figured we'd add them to separate PR's was just hoping to make a note of them so that we can add them to tasks etc. This is great, I've approved.🎉

**Comment by ludwiczakpawel** [comment]
@rhymes thanks, I'll take a look at these issues!

As for contrast of the circle icon... 🤷‍♂️ :P 

---

Re: test... I don't really know what I'm looking at. I mean, I don't have much experience with _tests_ generally... But I'm more than happy to be educated... Also, @benhalpern mentioned something about tests too, so I guess it's related..?

**Comment by joshpuetz** [comment]
Run it locally in Safari and on the iOS simulator and found a display issue: on mobile the reactions indictor isn't showing if there are zero reactions, but it does show on desktop:

DEV_local__Community_👩💻👨💻
iPhone_11_—_13_3

**Comment by ludwiczakpawel** [comment]
@joshpuetz hmmm that's weird, I'll take a look! :)

File: app/assets/stylesheets/article-show.scss [hunk 12]
```diff
@@ -99,41 +99,23 @@ article {
```
Comment (benhalpern) [inline app/assets/stylesheets/article-show.scss]: On the show page, the "offscreen" additional articles are actually showing up due to a 1px change.

Screen Shot 2020-02-13 at 6 45 45 PM

Basically when the height of the window changes, flexbox cleverly knows how to cut off whole elements instead of clipping them, and it does it this way and it's causing this funny visual thing.

**Comment by nickytonline** [comment]
@ludwiczakpawel, I'm loving the homepage improvements. Still going through the PR. Given that we have themeing, I'm wondering if we can make the feed pop more if articles have a background colour that isn't as close to the background colour of the page. I took an extreme example here, but I made article content areas `AntiqueWhite` and it brings the feed into focus more. Question one, what do you think about having more differentiation between page bg colour and card content colour and two, if you're for it, what would you suggest for card content bg colour given that we have themes?

image

**Comment by nickytonline** [comment]
> In addition, I don't know what your thoughts are on this and perhaps this is a discussion for a slack channel but I think using rem or even em instead of hardcoded pixel values everywhere in the css will at least start a journey towards accessibility.

@Ridhwana, rem and em are the way to go for accessiblity, zoom, user specific browser settings. Most of the app's CSS uses pixels and absolute positioning heavily, so it's going to be baby steps to move towards rem/em, but wherever we can start doing it, the better.

File: app/assets/javascripts/utilities/buildArticleHTML.js.erb (hunk 3 above)
Comment (nickytonline) [reply app/assets/javascripts/utilities/buildArticleHTML.js.erb]: These changes will also need to be ported to the new Preact components that haven't been merged yet.

File: app/assets/stylesheets/articles.scss [hunk 13]
```diff
@@ -4,297 +4,74 @@
 @import 'variables';
 @import '_mixins';
 
+
 .home {
   position: relative;
-  min-height: 440px;
-  margin: auto;
+  min-height: 440px; // TODO: why?
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: This might have been added back in the day when there was less content?

File: app/assets/stylesheets/articles.scss [hunk 14]
```diff
@@ -4,297 +4,74 @@
...
-    width: 100%;
+    max-width: 100%;
     position: relative;
     -webkit-tap-highlight-color: rgba(0, 0, 0, 0); //Invisible black
-    max-width: 750px;
-    padding-bottom: 30px;
-    @media screen and (min-width: 750px) {
-      margin-bottom: 30px;
-    }
+    box-sizing: border-box;
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: Just a general comment about `box-sizing`. We should probably have everything default to this as it simplifies calculating widths/heights.

File: app/assets/stylesheets/articles.scss [hunk 15]
```diff
@@ -4,297 +4,74 @@
...
-        &.selected {
-          @include themeable(
-            background,
-            theme-container-accent-background,
-            lighten($bold-blue, 8%)
-          );
-          @include themeable(color, theme-color, white);
-        }
-      }
+      max-width: calc(100% - 280px);
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: Do we know what the `280px` is? Is it potentially just margin `140px` on each side?

File: app/assets/stylesheets/articles.scss [hunk 16]
```diff
@@ -4,297 +4,74 @@
...
-          border: 0;
-          font-family: $helvetica-condensed;
-          font-stretch: condensed;
-          font-weight: 600;
-          cursor: pointer;
-        }
-      }
+
+    @media screen and (min-width: 1120px) {
+      max-width: calc(100% - 280px - 280px);
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]:
```suggestion
      max-width: calc(100% - 560px);
```

File: app/assets/stylesheets/articles.scss [hunk 17]
```diff
@@ -4,297 +4,74 @@
...
       }
     }
+
     a {
       @include themeable(color, theme-container-color, $black);
     }
-    a.small-pic-link-wrapper,
-    .placeholder-div {
+
+    .placeholder-div { // TOOD: what is this?
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: There is an empty `<div />` for related to this class in a few files, but it's a `<div />`, so `display: block` is redundant as that's the default for a `<div />` and even the width is redundant as block elements take up the whole width by default.

image

File: app/assets/stylesheets/articles.scss [hunk 18]
```diff
@@ -395,60 +163,38 @@
...
-          padding: 20px 0px 23px;
-        }
-        @media screen and (min-width: 1400px) {
-          padding: 23px 0px 27px;
-        }
+        margin: 16px 0 0 16px;
+        border-radius: 50%;
         img,
-        .color {
+        .color { // TODO: what is .color for?
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: I don't think it's used based on what I saw in the codebase, but if you're not positive, just leave the TODO.

File: app/assets/stylesheets/articles.scss [hunk 19]
```diff
@@ -523,44 +265,36 @@
...
+
         &:hover {
           opacity: 0.9;
           text-decoration: underline;
         }
       }
       .tags {
         margin-top: 0px;
-        font-size: 14px;
+        font-size: 0.8em;
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: I'm assuming you want this `font-size` relative to the parent? If not, use `rem`.

File: app/assets/stylesheets/articles.scss [hunk 20]
```diff
@@ -592,166 +321,105 @@
...
-        height: 50px;
-        width: 50px;
-      }
       &:hover {
         opacity: 0.9;
       }
     }
     .featured-user-name {
       font-weight: bold;
-      margin-top: -101px;
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: Any idea why this was `-101px`?

File: app/assets/stylesheets/articles.scss [hunk 21]
```diff
@@ -953,123 +604,16 @@
...
-          font-weight: 500;
-          font-size: 0.9em;
-          @include themeable(
-            background,
-            theme-background,
-            darken($light-gray, 4%)
-          );
-          a {
-            @include themeable(color, theme-secondary-color, $black);
+            -webkit-overflow-scrolling: touch;
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]: We should really install [autoprefixer](https://www.npmjs.com/package/autoprefixer)

File: app/assets/stylesheets/articles.scss [hunk 22]
```diff
@@ -1773,6 +1285,149 @@
...
+
+
+
+
+
+
+
+
+////////////////////////////////////////////////
+// TO COMPONETIZE //////////////////////////////
```
Comment (nickytonline) [inline app/assets/stylesheets/articles.scss]:
```suggestion
// TO COMPONENTIZE //////////////////////////////
```

File: app/assets/stylesheets/preact/article-form.scss [hunk 23]
```diff
@@ -41,20 +42,21 @@
   width: 97%;
   max-width: 860px;
   padding: 0px 6px;
-  background: white;
-  background: var(--theme-container-background, white);
+  @include dev-card;
+  // background: white;
```
Comment (nickytonline) [inline app/assets/stylesheets/preact/article-form.scss]: If the code is no longer used, delete it. We have git.

File: app/assets/stylesheets/articles.scss (hunk 13 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: Yea, I left some `// TODO` comments for myself to figure it out later, kind of notes :D 

Re this one: yes, Ben mentioned that too. He also mentioned it may not be needed anymore, but I'd rather clean it up in separate PR so I can test and be sure removing it won't do any harm.

File: app/assets/stylesheets/articles.scss (hunk 14 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: hahah, I [left a comment](https://github.com/thepracticaldev/dev.to/blob/b8dcb0176d3cf2872ca4e2c17b232af61dbb3e1f/app/assets/stylesheets/articles.scss#L215) in that very file to myself that this should be a global thing. 

So yea, 100% agree - but again: I didn't want to refactor everything in that PR. This change would actually cause a lot of harm right away :)

File: app/assets/stylesheets/articles.scss (hunk 15 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: It's the sidebar width. Previously it was some more/less random. I unified it to be 280px on both sidebars for now. This part of code basically defines the layout of the page and there's lots of things that could be improved (go with flexbox/grid instead of floats, isolate layout-defining properties to be separated from rest of the code, be smarter in terms actually defining all these sizes).

File: app/assets/stylesheets/articles.scss (hunk 16 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: I've written it that way to make it explicit it's related to two sidebar width. `560px` feels more random without any context. But I guess I can just add a comment there :)

File: app/assets/stylesheets/articles.scss (hunk 17 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: Yes, I was looking for that as well. I found commit form several months ago adding this thing. You're right CSS doesn't make much sense here but so do the element in DOM.  So I'd rather dig a bit more and make sure I can remove both: html & css.

File: app/assets/stylesheets/articles.scss (hunk 20 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: This is what you end up with when you want to position everything absolutely :D  I'm pretty sure this is the reason for that. We have/had more margins like that in feed card. 

Hopefully, we can rewrite the frontend code when working on new feed cards.

File: app/assets/stylesheets/articles.scss (hunk 21 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/articles.scss]: I think we do have it, no? Or at least something that automatically adds vendor prefixes...

However, for this particular property, autoprefixer won't work. It's only webkit property and it doesn't really exist in official CSS spec. The reason why I added it is to implement smooth scrolling on iOS browsers. Everything that has `overflow: scroll` CSS property will not scroll smoothly on iOS browser. Honestly, it's not required but just a nice touch.

File: app/assets/stylesheets/preact/article-form.scss (hunk 23 above)
Comment (ludwiczakpawel) [reply app/assets/stylesheets/preact/article-form.scss]: 100% agree. I'll clean it up a little.

File: app/assets/javascripts/initializers/initializeBaseUserData.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/javascripts/initializers/initializeBaseUserData.js]: Thanks, TIL. (I'm not really good at JS, with these changes I've been mainly updating existing code).

Can you look at that whole function? Does it really make sense here? Feels like we're adding some unnecessary lines of code but I might be wrong.  Also, this whole user profile section could probably become a preact component, no?

**Comment by ludwiczakpawel** [comment]
@rhymes , @benhalpern that weird extra border should now be gone.

@rhymes labels in dark mode should now be fixed.

@joshpuetz the missing reactions are actually caused by something else but I don't know what. something related to lazy loading, I guess? https://d.pr/v/tEYeSQ when I scroll down, page DOM is being reloaded (?) and then it doesn't have reactions count anymore... cc @benhalpern  it happens on prod as well.

**Comment by rhymes** [comment]
@ludwiczakpawel I can confirm those weird borders have disappeared, great catch!

**Comment by joshpuetz** [comment]
@ludwiczakpawel Oh wow, good catch on the missing reactions in prod! That should not block this then!

File: app/views/articles/_single_story.html.erb [hunk 24]
```diff
@@ -14,7 +14,7 @@
   <% end %>
   <a href="/<%= story.cached_user.username %>" class="small-pic-link-wrapper">
     <div class="small-pic">
-      <img src="<%= ProfileImage.new(story.cached_user).get(width: 90) %>" alt="<%= story.cached_user.username %> profile" />
+      <img src="<%= ProfileImage.new(story.cached_user).get(width: 90) %>" alt="<%= story.cached_user.username %> profile" width="40" height="40" />
```
Comment (rhymes) [inline app/views/articles/_single_story.html.erb]: I was wondering, but maybe you chose this because the height would be too short, if the width of the `<img` tag is 40, why not also ask for a `width: 40` in the method? Could you try and see what happens?

`ProfileImage.new(story.cached_user).get(width: 40)`

File: app/views/articles/index.html.erb [hunk 25]
```diff
@@ -95,7 +95,7 @@
               </h3>
             </div>
             <a href="/<%= @featured_story.cached_user.username %>" class="featured-profile-button">
-              <img class="featured-profile-pic" src="<%= ProfileImage.new(@featured_story.cached_user).get(width: 90) %>" alt="<%= @featured_story.title %>" />
+              <img class="featured-profile-pic" src="<%= ProfileImage.new(@featured_story.cached_user).get(width: 90) %>" width="40" height="40" alt="<%= @featured_story.title %>" />
```
Comment (rhymes) [inline app/views/articles/index.html.erb]: I was wondering, but maybe you chose this because the height would be too short, if the width of the <img tag is 40, why not also ask for a width: 40 in the method? Could you try and see what happens?

`ProfileImage.new(story.cached_user).get(width: 40)`

File: app/views/articles/_single_story.html.erb (hunk 24 above)
Comment (ludwiczakpawel) [reply app/views/articles/_single_story.html.erb]: But I actually want to have twice as big image so it looks sharp and crispy on retina when resized manually to 40x40.

File: app/views/articles/index.html.erb (hunk 25 above)
Comment (benhalpern) [reply app/views/articles/index.html.erb]: I don't know what the ideal answer is, but if we're already fetching `width: 90` elsewhere, it means that image will be cached in more places, so it's more efficient to keep fetching that.

Width 90 is sort of our standard small profile size.

**Comment by benhalpern** [comment]
> get rid of loading glitch

This is now looking pretty great to me. I'll keep snooping around, but I think it's in a pretty good place.

**Comment by ludwiczakpawel** [comment]
thanks @nickytonline 

the listings margin was an easy fix so I included it in this PR. consider it done.

**Comment by benhalpern** [comment]
@ludwiczakpawel the failing tests will all need to be updated to reflect the new things they are looking for on the page.

**Comment by ludwiczakpawel** [comment]
@benhalpern I don't really have too much knowledge about that... I was going to ping Rhymes the other day about it but haven't had a chance to actually sync up yet :(
